main.tf

```diff
@@ -34,17 +40,31 @@ locals {
   validate_resolver_type_input = (var.resolver_type != null && var.update_delegated_resolver == true) ? tobool("var.resolver_type cannot be set if var.update_delegated_resolver is set to true. Only one type of resolver can be created by VPC.") : true
 }
 
+##############################################################################
+# Check if existing vpc id is passed
+##############################################################################
+
+data "ibm_is_vpc" "vpc" {
+  count      = var.create_vpc == false ? 1 : 0
+  identifier = var.existing_vpc_id
+}
+
+locals {
+  vpc_id   = var.create_vpc ? ibm_is_vpc.vpc[0].id : var.existing_vpc_id
+  vpc_data = var.create_vpc ? ibm_is_vpc.vpc[0] : data.ibm_is_vpc.vpc[0]
+}
+
 ##############################################################################
 # Create new VPC
 ##############################################################################
 
 resource "ibm_is_vpc" "vpc" {
   count          = var.create_vpc == true ? 1 : 0
-  name           = var.prefix != null ? "${var.prefix}-${var.name}-vpc" : "${var.name}-vpc"
+  name           = "${var.prefix != null && var.name != null ? "${var.prefix}-${var.name}" : var.prefix != null ? var.prefix : var.name}-vpc"
```

Comment:
This logic still does not provide the ability to have full control over the VPC name - suggest to add optional `vpc_name` variable